puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/test_foreman.groovy

```diff
@@ -29,14 +29,40 @@ pipeline {
                                 configureRVM(env.RUBY_VER, env.GEMSET)
                                 databaseFile(gemset(env.GEMSET))
                                 configureDatabase(env.RUBY_VER, env.GEMSET)
-                                withRVM(['npm install'], env.RUBY_VER, env.GEMSET)
                             }
                         }
                         stage("unit-tests-2.5-postgres") {
                             steps {
                                 withRVM(['bundle exec rake jenkins:unit TESTOPTS="-v" --trace'], env.RUBY_VER, env.GEMSET)
                             }
                         }
+                    }
+                    post {
+                        always {
+                            cleanup(env.RUBY_VER, env.GEMSET)
+                            deleteDir()
+                        }
+                    }
+                }
+                stage('ruby-2.5-postgres-integrations') {
+                    agent { label 'fast' }
+                    environment {
+                        RUBY_VER = '2.5'
+                        GEMSET = 'ruby-2.5-postgres'
+                    }
+                    stages {
+                        stage("setup-2.5-postgres") {
+                            steps {
+                                git url: 'https://github.com/theforeman/foreman', branch: foreman_branch
+                                script {
```

Comment:
I don't think this step is required, we already get the `commit_hash` in the regular PostgreSQL+Ruby2.5 run.

Reply:
But it might run on another slave..?

Reply:
Doesn't matter because the commit_hash is an input parameter to the whole job.

Reply:
should i remove the setup stage then?

Reply:
removed the script section